From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: 404Setup <153366651+404Setup@users.noreply.github.com>
Date: Mon, 8 Jul 2024 20:09:11 +0800
Subject: [PATCH] Divine: Fix MC-2025

Original code by NONPLAYT
You can find the original code on https://github.com/DivineMC/DivineMC

Original post on Mojira: https://bugs.mojang.com/browse/MC-2025
Fix taken from Reddit: https://redd.it/8pgd4q

diff --git a/src/main/java/net/minecraft/world/entity/Entity.java b/src/main/java/net/minecraft/world/entity/Entity.java
index 9c3b819b7d0bb3b6f1ceb0f9f341074493514fc5..11a283dad82f98fd33f83b69ff77ee037b8cef42 100644
--- a/src/main/java/net/minecraft/world/entity/Entity.java
+++ b/src/main/java/net/minecraft/world/entity/Entity.java
@@ -2619,6 +2619,17 @@ public abstract class Entity implements SyncedDataHolder, Nameable, EntityAccess
 
             nbttagcompound.put("Vine.Storage", vineStorage); // Vine - vine data bundler
             nbttagcompound.put("Leaves.Data", leavesData); // Leaves - leaves ex data
+
+            // DivineMC start - Fix MC-2025
+            // This fix was taken from Reddit: https://redd.it/8pgd4q
+            AABB boundingBox = getBoundingBox();
+            ListTag boundingBoxList = new ListTag();
+            for (double coord : new double[]{boundingBox.minX, boundingBox.minY, boundingBox.minZ, boundingBox.maxX, boundingBox.maxY, boundingBox.maxZ}) {
+                boundingBoxList.add(DoubleTag.valueOf(coord));
+            }
+            nbttagcompound.put("DivineMC.BoundingBox", boundingBoxList);
+            // DivineMC end
+
             return nbttagcompound;
         } catch (Throwable throwable) {
             CrashReport crashreport = CrashReport.forThrowable(throwable, "Saving entity NBT");
